hera_pspec/uvpspec.py

```diff
@@ -160,6 +164,103 @@ def get_integrations(self, key, *args):
 
         return self.integration_array[spw][blpairts, pol]
 
+    def get_dlys(self, key):
+        """
+        Get array of delays given a spectral window selection.
+
+        Parameters
+        ----------
+        key : int, or tuple with integer
+            Spectral window selection
+
+        Returns
+        -------
+        dlys : float ndarray, contains delays in nanosec of pspectra given spw
+        """
+        indices = self.spw_to_indices(key)
+        return self.dly_array[indices]
+
+    def get_blpair_seps(self):
+        """
+        For each baseline-pair, get the average baseline separation in ENU frame in meters.
+
+        Returns blp_avg_sep
+        -------
+        blp_avg_sep : float ndarray, shape=(Nblpairts,)
+        """
+        # get list of bl separations
+        bl_vecs = self.get_ENU_bl_vecs()
+        bls = self.bl_array.tolist()
+        blseps = np.array(map(lambda bl: np.linalg.norm(bl_vecs[bls.index(bl)]), self.bl_array))
+
+        # iterate over blpair_array
+        blp_avg_sep = []
+        for blp in self.blpair_array:
+            an = self.blpair_to_antnums(blp)
+            bl1 = self.antnums_to_bl(an[0])
+            bl2 = self.antnums_to_bl(an[1])
+            blp_avg_sep.append(np.mean([blseps[bls.index(bl1)], blseps[bls.index(bl2)]]))
+
+        return np.array(blp_avg_sep)
+
+    def get_kvecs(self, spw, little_h=True):
+        """
+        Get cosmological wavevectors for power spectra given an adopted cosmology.
+
+        Parameters
+        ----------
+        spw : int, choice of specral window
+
+        little_h : boolean, optional
+                Whether to have cosmological length units be h^-1 Mpc or Mpc
+                Default: h^-1 Mpc
+
+        Returns (k_perp, k_para)
+        -------
+        k_perp : float ndarray, containing perpendicular cosmological wave-vectors, shape=(Nblpairs,)
+        k_para : float ndarray, containing parallel cosmological wave-vectors, shape=(Ndlys given spw)
+
+        """
+        # assert cosmo
+        assert hasattr(self, 'cosmo'), "self.cosmo must exist to form cosmological wave-vectors. See self.add_cosmology()"
+
+        # calculate mean redshift of band
+        avg_z = self.cosmo.f2z(np.mean(self.freq_array[self.spw_to_indices(spw)]))
+
+        # get kperps
+        blpair_seps = self.get_blpair_seps()
+        k_perp = blpair_seps * self.cosmo.bl_to_kperp(avg_z, little_h=little_h)
+
+        # get kparas
+        k_para = self.get_dlys(spw) * self.cosmo.tau_to_kpara(avg_z, little_h=little_h)
+
+        return k_perp, k_para
+
+    def convert_to_deltasq(self, little_h=True):
+        """
+        Convert P(k) -> Delta^2(k) units.
```

Comment:
Can we be explicit about what the input and output units are, please? I think this will be useful for debugging later.